gcovr/utils.py

```diff
@@ -180,7 +180,12 @@ def __init__(self, regex, path_context=None):
 
 class Filter(object):
     def __init__(self, pattern):
-        self.pattern = re.compile(pattern)
+        cwd = os.getcwd()
+        is_fs_case_insensitive = os.path.exists(cwd.upper()) and os.path.exists(cwd.lower())
```

Comment:
I agree that using cwd is sensible. But there might be the case that `cwd = '/'` on Unix, which would falsely suggest a case-insensitive file system. In principle arbitrary non-alphabetic paths are possible, and I'm not sure to which degree this should be handled.

I'd suggest just a check `is_fs_case_insensitive = (cwd != '/') and ...`, ignoring other paths that are case-invariant because nearly all projects will use a sane folder layout in practice.

Reply:
I've used os.path.sep instead of '/'.